CONTRIBUTING.md

```diff
@@ -0,0 +1,151 @@
+
+Welcome!  Mypy is a community project that aims to work for a wide
+swath of Python users and Python codebases.  If you're trying Mypy on
+your Python code, your experience and what you can contribute is
+important to the project's success.
+
+
+Getting started, building, and testing
+--------------------------------------
+
+If you haven't already, take a look at the project's README file and
+the [Mypy documentation](http://mypy.readthedocs.org/en/latest/), and
+try adding type annotations to your file and type-checking it with Mypy.
+
+
+Discussion
+----------
+
+If you've run into behavior in Mypy you don't understand, or you're
+having trouble working out a good way to apply it to your code, or
+you've found a bug or would like a feature it doesn't have, we want to
+hear from you!
+
+Our main forum for discussion is the project's [GitHub issue
+tracker](https://github.com/python/mypy/issues).  This is the right
+place to start a discussion of any of the above or most any other
+topic concerning the project.
+
+We have an IRC channel, `#python-mypy` on irc.freenode.net.  This is
+lightly used, but some Mypy core developers are almost always present;
+feel free to find us there and we're happy to chat.  Substantive
+technical discussion will be directed to the issue tracker.
+
+#### Code of Conduct
+
+Everyone participating in the Mypy community, and in particular in our
+issue tracker, pull requests, and IRC channel, is expected to treat
+other people with respect and more generally to follow the guidelines
+articulated in the [Python Community Code of
+Conduct](https://www.python.org/psf/codeofconduct/).
+
+
+Submitting Changes
+------------------
+
+Even more excellent than a good bug report is a fix for a bug, or the
+implementation of a much-needed new feature.  We'd love to have your
+contributions.
+
+We use the usual GitHub pull-request flow, which may be familiar to
+you if you've contributed to other projects on GitHub.  For the mechanics
+of it, see [our git and GitHub workflow help page](http://www.mypy-lang.org/wiki/UsingGitAndGitHub),
+or [GitHub's own documentation](https://help.github.com/articles/using-pull-requests/).
+
+Anyone interested in Mypy may review your code.  One of the Mypy core
+developers will merge your pull request when they think it's ready.
+For every pull request, we aim to promptly either merge it or say why
+it's not yet ready; if you go a few days without a reply, please feel
+free to ping the thread with a new comment.
+
+At present the core developers are (alphabetically):
+* David Fisher (@ddfisher)
+* Jukka Lehtosalo (@JukkaL)
+* Greg Price (@gnprice)
+* Guido van Rossum (@gvanrossum)
+
+
+Preparing Changes
+-----------------
+
+Before you begin: if your change will be a significant amount of work
+to write, we highly recommend starting by opening an issue laying out
+what you want to do.  (This is good advice for all kinds of
+open-source projects in general.)  That lets a conversation happen
```

Comment:
Hmm, yeah, it does come across as somewhat didactic. What I really want out of this parenthetical is mainly to signal to people who *do* have some experience or know the standard advice that we don't think we're doing anything particularly different here -- like with the "Standard advice ... applies" sentence in the next graf. I'll attempt a rephrase.

The hyphenation is another example of a compound modifier: https://github.com/python/mypy/pull/1268#discussion_r55761466 .
The Wikipedia article "Open source", for example, scrupulously hyphenates the phrase wherever it appears as a modifier: https://en.wikipedia.org/wiki/Open_source